platform/android/display_server_android.cpp

```diff
@@ -976,3 +981,18 @@ void DisplayServerAndroid::set_icon(const Ref<Image> &p_icon) {
 bool DisplayServerAndroid::is_window_transparency_available() const {
 	return GLOBAL_GET_CACHED(bool, "display/window/per_pixel_transparency/allowed");
 }
+
+void DisplayServerAndroid::gl_window_make_current(DisplayServer::WindowID p_window_id) {
+	if (p_window_id == INVALID_WINDOW_ID) {
+		return;
+	}
+
+	if (egl_current_window_id == p_window_id) {
+		return;
+	}
+
+	GodotJavaWrapper *godot_java = OS_Android::get_singleton()->get_godot_java();
+	ERR_FAIL_NULL(godot_java);
+	godot_java->make_gl_window_current(p_window_id);
```

Comment:
I'm not sure this would actually work in the context of multithreaded rendering, because I think the managed thread on the Java side may be a different thread than the C++ thread making this call. So, we may need to completely move the `eglMakeCurrent()` to the C++ side, rather than making a JNI call to do it on the Java side

I'll need to test this, though :-)

Reply:
The Java logic is run on the caller thread unless a call to `runOnHostThread` is done. So this should work as expected.